lib/model.js

```diff
@@ -3085,14 +3086,23 @@ class Model {
 
             // Convert date fields to real date objects
             if (this.constructor._hasDateAttributes && this.constructor._isDateAttribute(key) && !!value && !value._isSequelizeMethod) {
-              if (!(value instanceof Date)) {
-                value = new Date(value);
-              }
-              if (!(originalValue instanceof Date)) {
-                originalValue = new Date(originalValue);
-              }
-              if (originalValue && value.getTime() === originalValue.getTime()) {
-                return this;
+              // Dont parse DATEONLY to new Date, keep them as string
+              if (this.rawAttributes[key].type instanceof DataTypes.DATEONLY) {
+                if (originalValue && originalValue === value) {
+                  return this;
+                } else {
+                  value = moment(value).format('YYYY-MM-DD');
```

Comment:
When we build instance, we need to keep `YYYY-MM-DD` there as well, it prevent build instance to have `Date`. Its a safe guard against case when user are passing `new Date` / `moment` to `DATEONLY`. This is also done in stringify, but stringify is only called when we are putting data from instance to query